alarm/utils.py

```diff
@@ -0,0 +1,112 @@
+import json
+import ssl
+
+import websocket
+from sentry_sdk import capture_exception
+from queue import Queue
+from alarm.models import Coin
+
+previous_close_price = None
+previous_candlestick = None
+current_prices = None
+previous_prices = None
+
+
+def get_binance_price_and_analized(currencies, intervals):
+    websocket.enableTrace(False)
+    sockets = [f'wss://stream.binance.com:9443/ws/{currency}@kline_{interval}' for currency, interval in
+               zip(currencies, intervals)]
+
+    ws_list = []
+    for socket in sockets:
+        ws = websocket.create_connection(socket, sslopt={'cert_reqs': ssl.CERT_NONE})
+        ws.on_message = on_message
+        ws.on_close = on_close
+        ws_list.append(ws)
+
+    try:
+        while True:
+            for ws in ws_list:
+                message = ws.recv()
+                global previous_close_price
+                global previous_candlestick
+                global current_prices
+                global previous_prices
+                # Parse the JSON message
+                json_message = json.loads(message)
+
+                # Extract the candlestick data from the JSON message
+                candlestick = json_message['k']
+
+                # Extract the close price from the most recent candlestick
+                current_close_price = float(candlestick['c'])
+
+                # Update the previous close price, if a previous candlestick is available
+                if previous_candlestick is not None:
+                    previous_close_price = float(previous_candlestick['c'])
+
+                # Store the current candlestick data for use in the next iteration
+                previous_candlestick = candlestick
+
+                # Store the current and previous close prices in a global variable
+                current_prices = current_close_price
+                previous_prices = previous_close_price
+                short_coin_name = candlestick['s']
+                analize_prices(current_prices, previous_prices)
+                on_message(ws, message, short_coin_name)
+    except KeyboardInterrupt:
+        for i, ws in enumerate(ws_list):
+            on_close(ws, '', close_msg=f' {currencies[i]}')
+            ws.close()
+    except Exception as e:
```

Comment:
Usually, we never catch Exception, we try to catch a specific exception. And " as err"

Reply:
++